lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp

```diff
@@ -188,22 +191,45 @@ LLVMBasedICFG::LLVMBasedICFG(ProjectIRDB &IRDB, CallGraphAnalysisType CGType,
       processFunction(F, *Res, FixpointReached);
     }
 
-    for (auto [CS, _] : IndirectCalls) {
-      FixpointReached &= !constructDynamicCall(CS, *Res);
+    if (S == Soundness::Soundy) {
```

Comment:
We should be as conservative as possible when dealing with soundness. Here, we completely ignore `Soundness::Sound` which may be dangerous
```suggestion
    if (S != Soundness::Unsound) {
```